src/Components/3DV/SceneView.tsx

```diff
@@ -224,8 +225,53 @@ function SceneView(props: ISceneViewProps, ref) {
     debugLog('debug', 'debug', 'SceneView Render');
     const url = modelUrl === 'Globe' ? globeUrl : modelUrl;
 
-    const sortMeshesOnLoad = () => {
+    const preProcessMeshesOnLoad = () => {
+        let uniqueNumber = 0;
+        //Initial loop to check for issues and cleanup meshes.
         for (const mesh of sceneRef.current.meshes) {
+            //Let's make sure that all mesh ids are unique
+            const matchingId = sceneRef.current.meshes.filter(
+                (x) => x.id == mesh.id
+            );
+            if (matchingId.length > 1) {
+                //Throw an error here
+                console.warn(
+                    'Loaded model contains objects with matching names. 3D Scenes Studio only supports 3d models with unique object names. Attempting to recover by forcing unique names...'
```

Comment:
I'd change "matching" to "duplicate". It's clearer IMO.

Reply:
Sounds good to me